core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -319,14 +325,97 @@ class WorkspaceService(
+          workspace.baseWorkspace,
+          Option(
+            samResource.authDomainGroups.map(groupName => ManagedGroupRef(RawlsGroupName(groupName.value)))
+          ),
+          useAttributes = options.attrSpecs.all || options.attrSpecs.attrsToSelect.nonEmpty,
+          workspace.getCloudPlatform
+        )
+
+      val canShare = options.anyPresent("canShare") {
+        accessLevel match {
+          case WorkspaceAccessLevels.Read  => samResource.hasRole(SamWorkspaceRoles.shareReader)
+          case WorkspaceAccessLevels.Write => samResource.hasRole(SamWorkspaceRoles.shareWriter)
+          case _                           => accessLevel > WorkspaceAccessLevels.Write
+        }
+      }
+      val canCompute: Option[Boolean] = options
+        .anyPresent("canCompute") {
+          workspace.getCloudPlatform.map {
+            case WorkspaceCloudPlatform.Azure => accessLevel >= WorkspaceAccessLevels.Write
+            case WorkspaceCloudPlatform.Gcp if accessLevel >= WorkspaceAccessLevels.Owner => true
+            case WorkspaceCloudPlatform.Gcp => samResource.hasRole(SamWorkspaceRoles.canCompute)
+          }
+        }
+        .flatten
+      WorkspaceListResponse(
+        accessLevel,
+        canShare,
+        canCompute,
+        workspaceDetails,
+        stats,
+        samResource.public.roles.nonEmpty || samResource.public.actions.nonEmpty,
+        Some(workspace.policies)
+      )
+    }
+
+    for {
+      workspaceResources <- samDAO.listUserResources(SamResourceTypeNames.workspace, ctx)
+
+      // filter out the resources that do not have any roles related to access levels
+      // also filter out any policy whose resourceId is not a UUID; these will never match a known workspace
+      accessLevelWorkspaceResources = workspaceResources.filter(resource =>
+        resource.allRoles.exists(role => WorkspaceAccessLevels.withRoleName(role.value).nonEmpty) &&
+          Try(UUID.fromString(resource.resourceId)).isSuccess
+      )
+      accessLevelWorkspaceUUIDs = accessLevelWorkspaceResources.map(resource => UUID.fromString(resource.resourceId))
+      submissionSummaryStats <- options.anyPresentFuture("workspaceSubmissionStats") {
+        workspaceRepository.listSubmissionSummaryStats(accessLevelWorkspaceUUIDs)
+      }
+      workspaces <- workspaceRepository.listWorkspacesByIds(accessLevelWorkspaceUUIDs, Option(options.attrSpecs))
+      highestAccessLevelByWorkspaceId = accessLevelWorkspaceResources.map { resource =>
+        resource.resourceId -> resource.allRoles.flatMap(role => WorkspaceAccessLevels.withRoleName(role.value)).max
+      }.toMap
+      workspaceSamResourceByWorkspaceId = accessLevelWorkspaceResources.map(r => r.resourceId -> r).toMap
+      aggregatedWorkspaces = new AggregatedWorkspaceService(workspaceManagerDAO)
+        .fetchAggregatedWorkspaces(workspaces, ctx)
+        // Filter out workspaces with no cloud contexts, logging cloud context exceptions
```

Comment:
We no longer log the cloud context exceptions (which is probably fine, but the comment should be changed).